lib/graphql/client.ex

```diff
@@ -38,23 +38,56 @@ defmodule BridgeEx.Graphql.Client do
     * `retry_options`: configures retry attempts. Takes the form of `[max_retries: 1, timing: :exponential]`
     * `log_options`: configures logging on errors. Takes the form of `[log_query_on_error: false, log_response_on_error: false]`.
   """
+  @deprecated "This call uses `Json.decode` with `keys: :atoms` which is discouraged as it dynamically creates atoms which are not garbage collected. Please use `call/5` with a decoder instead. For instance, you can use `Client.call(url, query, variables, &Utils.string_decoder/1, opts)` which will be the default behavior in the future."
+  @spec call(
+          url :: String.t(),
+          query :: String.t(),
+          variables :: map(),
+          opts :: Keyword.t()
+        ) :: bridge_response()
+  def call(url, query, variables, opts),
+    do: call(url, query, variables, :atoms, opts)
 
+  @doc """
+  Calls a GraphQL endpoint and decodes the response
+
+  ## Parameters
+
+    * `url`: URL of the endpoint.
+    * `query`: Graphql query or mutation.
+    * `variables`: variables for Graphql query or mutation.
+    * `decoder`: decoder for the response. Can be an atom (:strings, :atoms, :existing_atoms) for JSON parsing or a function which transforms a body into a {:ok, any} or {:error, any} tuple.
+    * `opts`: various options.
+
+  ## Options
+
+    * `options`: extra HTTP options to be passed to Telepoison.
+    * `headers`: extra HTTP headers.
+    * `encode_variables`: whether to JSON encode variables or not.
+    * `retry_options`: configures retry attempts. Takes the form of `[max_retries: 1, timing: :exponential]`
+    * `log_options`: configures logging on errors. Takes the form of `[log_query_on_error: false, log_response_on_error: false]`.
+  """
   @spec call(
           url :: String.t(),
           query :: String.t(),
           variables :: map(),
+          decoder :: (String.t() -> bridge_response()) | atom(),
           opts :: Keyword.t()
         ) :: bridge_response()
+  def call(url, query, variables, decoder, opts) when is_atom(decoder),
+    do: call(url, query, variables, json_decoder(decoder), opts)
+
   def call(
         url,
         query,
         variables,
+        decoder,
```

Comment:
I thought about this, TBH, but this change allows method deprecation. If the decoder is part of the options, it has to raise a warning (which is what I did in the graphql part), but I think that a deprecation is more "visible" when working from an IDE.

Reply:
Is there any way to check if a keyword list contains a key from the pattern match?

Reply:
I think we don't want to deprecate the method without the decoder parameter.
We'd like users to specify the decoding mode explicitly for now because we are going to change the default behavior in the future. I'd rather not force users to specify it anymore when `:strings` becomes the default decoder.

Reply:
> Maybe using a warning?

Yeah, that's fine. Once we change the default `decode_keys` option, we'll remove the warning.